src/braket_container.py

```diff
@@ -149,14 +150,39 @@ def kick_off_customer_script(entry_point : str) -> multiprocessing.Process:
         str_module, _, str_method = entry_point.partition(":")
         customer_module = importlib.import_module(str_module)
         customer_method = getattr(customer_module, str_method)
-        customer_code_process = multiprocessing.Process(target=customer_method)
+
+        process_kwargs = {"target": customer_method}
+
+        function_args = try_bind_hyperparameters_to_customer_method(customer_method)
+        if function_args is not None:
+            process_kwargs["kwargs"] = function_args
+
+        customer_code_process = multiprocessing.Process(**process_kwargs)
         customer_code_process.start()
     except Exception as e:
         log_failure_and_exit(f"Unable to run job at entry point {entry_point}\nException: {e}")
     return customer_code_process
 
 
-def join_customer_script(customer_code_process : multiprocessing.Process):
+def try_bind_hyperparameters_to_customer_method(customer_method: Callable):
+    hp_file = os.getenv("AMZN_BRAKET_HP_FILE")
+    with open(hp_file) as f:
+        hyperparameters = json.load(f)
+
+    try:
+        inspect.signature(customer_method).bind(**hyperparameters)
+        annotations = inspect.getfullargspec(customer_method).annotations
+        function_args = {}
+        for param in hyperparameters:
+            function_args[param] = annotations.get(param, str)(
+                hyperparameters[param]
+            )
+        return function_args
+    except TypeError:
+        pass
```

Comment:
we could move things around here to isolate the try/except logic

```suggestion
    try:
        can_bind = inspect.signature(customer_method).bind(**hyperparameters)
    except TypeError:
        pass
    if can_bind:
        annotations = inspect.getfullargspec(customer_method).annotations
        function_args = {}
        for param in hyperparameters:
            function_args[param] = annotations.get(param, str)(
                hyperparameters[param]
            )
        return function_args
```